### PR TITLE
[ENH] in `BaseRegressor`, allow `y` to be 1D `pd.DataFrame`

### LINKED ISSUE #5177: [BUG] BaseRegressor should allow one-column `pd.DataFrame` as `y`
**Describe the bug**

The BaseRegressor's fit method performs the type coercion only for X and not for y. Thus calling a Regressor with a one-dimensional data frame as y fails.

**To Reproduce**

```python
from sktime.regression.distance_based import KNeighborsTimeSeriesRegressor
import numpy as np

knn = KNeighborsTimeSeriesRegressor()

# The following statement works
knn.fit(pd.DataFrame(np.random.normal(0, 1, (100, 5)), index=pd.MultiIndex.from_product([range(50), range(2)])),
        pd.Series(np.random.normal(0, 1, 50))
        )

# The following statement fails  
knn.fit(pd.DataFrame(np.random.normal(0, 1, (100, 5)), index=pd.MultiIndex.from_product([range(50), range(2)])),
        pd.DataFrame(np.random.normal(0, 1, 50))
        )
```

**Expected behavior**

Both statements should work and the one-dimensional y is converted to a series internally.

### PR DESCRIPTION
**Pull request by mdsaad2305**
#### Reference Issues/PRs
fixes #5177 

#### What does this implement/fix? Explain your changes.
added functionality to accommodate 1-D pandas DataFrame and fixed import statements where necessary

### CODE REVIEW

File: sktime/dists_kernels/dtw/_dtw_python.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 
-from sktime.distances import pairwise_distance
+from sktime.distances._distance import pairwise_distance
```
Comment (yarnabrina) [inline sktime/dists_kernels/dtw/_dtw_python.py:9]: Why is this required? It's being reexported in the `__init__` moduke, is this insufficient?

https://github.com/sktime/sktime/blob/d4a0dc3f9a622b673bc385a83a8271f5fe99b81c/sktime/distances/__init__.py#L50

File: sktime/dists_kernels/edit_dist.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 
-from sktime.distances import pairwise_distance
+from sktime.distances._distance import pairwise_distance
```
Comment (yarnabrina) [inline sktime/dists_kernels/edit_dist.py:9]: Same as above.

File: sktime/regression/distance_based/_time_series_neighbors.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 from sklearn.neighbors import KNeighborsRegressor
 
-from sktime.distances import pairwise_distance
+from sktime.distances._distance import pairwise_distance
```
Comment (yarnabrina) [inline sktime/regression/distance_based/_time_series_neighbors.py:16]: Same as above.

**Review by yarnabrina — APPROVED** [review]
Looks good to me, but let's wait for @benHeid who created the bug report.

Also, if you add a test case for dataframe (positive with one column, negative with two columns), it'd be great.

**Comment by mdsaad2305** [comment]
Is there any specific place I need to put the test cases in?

**Comment by fkiraly** [comment]
I would also approve, although I think we want to replace the `y` target logic, mid-term, with a `check_is_scitype` for the `"Table"` scitype. In-principle, we should allow multivariate `y` - and broadcast in case the regressor is univariate.

Though that might be better to do separately, and accept this as a quick fix.

**Comment by benHeid** [comment]
Sorry for the late response... I was blocked with a lot of different stuff...

This PR looks good to me. Should we add the example from the issue as a test?

**Comment by fkiraly** [comment]
> This PR looks good to me. Should we add the example from the issue as a test?

Yes, that would be a good idea. Merging in prep for release though, so separate PR?
